streampipes-extensions/streampipes-sinks-databases-jvm/src/main/java/org/apache/streampipes/sinks/databases/jvm/milvus/MilvusSink.java

```diff
@@ -0,0 +1,380 @@
+        .withAssets(ExtensionAssetType.DOCUMENTATION, ExtensionAssetType.ICON).
+        category(DataSinkType.DATABASE)
+        .requiredTextParameter(Labels.withId(MILVUS_URI_KEY))
+        .requiredTextParameter(Labels.withId(MILVUS_TOKEN_KEY), "root:Milvus")
+        .requiredTextParameter(Labels.withId(MILVUS_DBNAME_KEY))
+        .requiredTextParameter(Labels.withId(DATABASE_REPLICA_NUMBER_KEY), "2")
+        .requiredTextParameter(Labels.withId(COLLECTION_NAME_KEY))
+        .requiredTextParameter(Labels.withId(PRIMARY), "id")
+        .requiredIntegerParameter(Labels.withId(DIMENSION), 2)
+        .requiredStream(StreamRequirementsBuilder
+            .create()
+            .requiredPropertyWithUnaryMapping(EpRequirements.anyProperty(),
+                Labels.withId(VECTOR_KEY),
+                PropertyScope.NONE)
+            .build())
+        .requiredSingleValueSelection(Labels.withId(INDEX),
+             Options.from(INDEX_MAP.keySet().toArray(new String[0])))
+        .requiredSingleValueSelection(Labels.withId(METRIC_TYPE),
+             Options.from(METRIC_TYPE_MAP.keySet().toArray(new String[0])))
+        .build();
+  }
+
+  @Override
+  public void onInvocation(SinkParams parameters,
+                           EventSinkRuntimeContext runtimeContext) throws SpRuntimeException {
+    var extractor = parameters.extractor();
+    final String uri = extractor.singleValueParameter(MILVUS_URI_KEY, String.class);
+    final String token = extractor.singleValueParameter(MILVUS_TOKEN_KEY, String.class);
+
+    ConnectConfig connectConfig = ConnectConfig.builder()
+        .uri(uri)
+        .token(token)
+        .build();
+
+    PoolConfig poolConfig = PoolConfig.builder()
+        .maxIdlePerKey(10) // max idle clients per key
+        .maxTotalPerKey(20) // max total(idle + active) clients per key
+        .maxTotal(100) // max total clients for all keys
+        .maxBlockWaitDuration(Duration.ofSeconds(5L)) // getClient() will wait 5 seconds if no idle client available
+        .minEvictableIdleDuration(Duration.ofSeconds(10L)) // if number
+        .build();
+
+    try {
+      pool = new MilvusClientV2Pool(poolConfig, connectConfig);
+      client = pool.getClient("client_name");
+      //create a dataBase
+      final String dbName = parameters.extractor().singleValueParameter(MILVUS_DBNAME_KEY, String.class);
+      final String dbReplicaNum =
+              parameters.extractor().singleValueParameter(DATABASE_REPLICA_NUMBER_KEY, String.class);
+      Map<String, String> properties = new HashMap<>();
+      properties.put(Constant.DATABASE_REPLICA_NUMBER, dbReplicaNum);
+      try {
+        CreateDatabaseReq createDatabaseReq = CreateDatabaseReq.builder()
+            .databaseName(dbName)
+            .properties(properties)
+            .build();
+        client.createDatabase(createDatabaseReq);
+        client.useDatabase(dbName);
+      } catch (Exception ignored) {
+        //todo add log
```

Comment:
Can you add some logging here? You can throw a `SpRuntimeException` so that any problems when creating the database connection will show up in the UI.